File: src/main/java/com/student/service/rag/AdminService.java
package com.student.service.rag;

import com.student.dto.ApiResponse;
import com.student.dto.rag.AdminStatsVO;
import com.student.dto.rag.ApproveRequest;
import com.student.dto.rag.UnmappedTermVO;

import java.util.List;

/**
 * 管理员审核服务接口
 * 提供未映射词条的分页查询、审核通过、拒绝、统计等功能
 *
 * @author 系统
 * @version 1.0
 */
public interface AdminService {

    /**
     * 分页获取待审核词条
     *
     * @param page 页码
     * @param size 每页大小
     * @param status 筛选状态（PENDING/APPROVED/REJECTED，null表示全部）
     * @return 分页结果
     */
    ApiResponse.Pagination<List<UnmappedTermVO>> getPendingTerms(int page, int size, String status);

    /**
     * 单个审核通过
     *
     * @param id 词条ID
     * @param standardTerm 标准术语
     * @param reviewer 审核人用户名
     */
    void approveTerm(Long id, String standardTerm, String reviewer);

    /**
     * 批量审核通过
     *
     * @param requests 审核请求列表
     * @param reviewer 审核人用户名
     * @return 成功数量
     */
    int batchApprove(List<ApproveRequest> requests, String reviewer);

    /**
     * 拒绝词条（软删除）
     *
     * @param id 词条ID
     * @param reason 拒绝原因
     * @param reviewer 审核人用户名
     */
    void rejectTerm(Long id, String reason, String reviewer);

    /**
     * 获取审核统计数据
     *
     * @return 统计数据
     */
    AdminStatsVO getStats();
}

File: frontend/src/api/services/admin.service.ts
import request from '../request'

/** 未映射词条项 */
export interface UnmappedTermItem {
  id: number
  term: string
  contextQuery: string
  guessedEntityType: string
  llmGuess?: string
  llmConfidence?: number
  occurrenceCount: number
  status: string
  reviewer?: string
  reviewNote?: string
  firstSeenAt: string
  lastSeenAt: string
  entityTypeLabel?: string
  reviewerName?: string
}

/** 分页响应 */
export interface PaginationResponse<T> {
  data: T
  total: number
  page: number
  size: number
  totalPages: number
}

/** 审核请求 */
export interface ApproveRequest {
  id: number
  standardTerm: string
  entityType?: string
  snomedConceptId?: string
}

/** 统计数据 */
export interface AdminStats {
  pendingCount: number
  approvedTodayCount: number
  unmappedRate: number
  alertThreshold: number
  alertTriggered: boolean
  totalAlertCount: number
  topUnmappedTerms: string[]
}

class AdminService {
  /** 分页获取待审核词条 */
  // silent: 页面自动加载的读操作建议传 { silent: true }，由视图呈现错误态，避免与全局弹窗重复提示
  async getUnmappedTerms(
    params: {
      page?: number
      size?: number
      status?: string
    } = {},
    options?: { silent?: boolean }
  ): Promise<PaginationResponse<UnmappedTermItem[]>> {
    const { page = 1, size = 20, status } = params
    const query = new URLSearchParams()
    query.append('page', page.toString())
    query.append('size', size.toString())
    if (status) query.append('status', status)

    // 失败由拦截器统一弹窗/reject，成功直接返回业务数据
    const response = await request.get<{
      data: {
        data: UnmappedTermItem[]
        total: number
        page: number
        size: number
        totalPages: number
      }
    }>(`/admin/unmapped-terms?${query.toString()}`, { silent: options?.silent })
    return response.data.data
  }

  /** 获取统计数据 */
  async getStats(options?: { silent?: boolean }): Promise<AdminStats> {
    const response = await request.get<{ data: AdminStats }>('/admin/unmapped-terms/stats', {
      silent: options?.silent
    })
    return response.data.data
  }

  /** 单个审核通过 */
  async approveTerm(id: number, standardTerm: string): Promise<void> {
    await request.post(`/admin/unmapped-terms/${id}/approve`, { standardTerm })
  }

  /** 批量审核通过 */
  async batchApprove(requests: ApproveRequest[]): Promise<{ total: number; success: number; failed: number }> {
    const response = await request.post<{
      data: { total: number; success: number; failed: number }
    }>('/admin/unmapped-terms/batch-approve', requests)
    return response.data.data
  }

  /** 拒绝词条 */
  async rejectTerm(id: number, reason: string = ''): Promise<void> {
    await request.post(`/admin/unmapped-terms/${id}/reject`, { reason })
  }

  /** 获取未映射率 */
  async getUnmappedRate(): Promise<AdminStats> {
    const response = await request.get<{ data: AdminStats }>('/admin/stats/unmapped-rate')
    return response.data.data
  }
}

export const adminService = new AdminService()
